### PR TITLE
Added helper to clone provisioning templates

### PR DESCRIPTION
**Pull request by oshtaier**
```
>>> template = entities.ConfigTemplate().create(True)
>>> new_template = entities.ConfigTemplate(id=template.id).clone(data={u'name': 'test'})
>>> new_template
{u'locked': False, u'name': u'test', u'default': False, u'created_at': u'2016-06-21T10:36:04.781Z', u'updated_at': u'2016-06-21T10:36:04.781Z', u'snippet': False, u'template_kind_id': 1, u'os_family': None, u'template': u'\U0002636c\U00025106\u99e7\u0928\U0002f994\u8e91\uc992\u53af\ub349\u8192\ub57e', u'job_category': u'Miscellaneous', u'vendor': None, u'description_format': None, u'id': 398, u'provider_type': None}
```

### CODE REVIEW

File: nailgun/entities.py [hunk 1]
```diff
@@ -1000,6 +1003,22 @@ def build_pxe_default(self, synchronous=True, **kwargs):
...
+            ``True``. Immediately return the server's response otherwise.
+        :param kwargs: Arguments to pass to requests.
+        :returns: The server's response, with all JSON decoded.
+        :raises: ``requests.exceptions.HTTPError`` If the server responds with
+            an HTTP 4XX or 5XX message.
+        """
+        kwargs = kwargs.copy()  # shadow the passed-in kwargs
+        kwargs.update(self._server_config.get_client_kwargs())
+        response = client.post(self.path('clone'), **kwargs)
+        return _handle_response(response, self._server_config, synchronous)
```
Comment (abalakh) [inline nailgun/entities.py]: I'd propose to do it like that:
```python
        kwargs = kwargs.copy()  # shadow the passed-in kwargs
        kwargs.update(self._server_config.get_client_kwargs())
        response = client.post(self.path('clone'), **kwargs)
        result = _handle_response(response, self._server_config, synchronous)
        if 'id' not in result:
            raise APIResponseError(
                'Server have not returned cloned provision template ID')
        return ConfigTemplate(id=result['id']).read()
```
In this case you'll receive not just some dictionary, but nailgun entity, like we usually receive for any basic operation like create/update.
So your example will become even simplier:
```python
>>> template = entities.ConfigTemplate().create(True)
>>> new_template = template.clone(data={'name': 'test111'})
>>> new_template
nailgun.entities.ConfigTemplate(nailgun.config.ServerConfig(url=u'https://example.com', verify=False, auth=(u'admin', u'password')), snippet=True, locked=False, template_kind=None, operatingsystem=[], template_combinations=[], location=[], template=u'\U000266d3\u4096\U00029041\U00024c9c\u06bb\U00029f07\u68f8\ub567\U0002176c\u3ccb\u6676\uc0c5\U00027a8f\U00021bf6\U000270d7\U000233ff\U00028075\ud1dc\U0001312d\u3c06\uc58e\ub1bc\u9c0c\ucfe0', organization=[], audit_comment=None, id=400, name=u'test111')
```
But as it's a helper and not basic operation, and we already have `ContentView.copy()` which is not following this approach (by accident or on purpose?) this suggestion is debatable.
@Ichimonji10 , your vision, please?

**Comment by oshtaier** [comment]
Don't mind to use proposed solution

File: nailgun/entities.py [hunk 2]
```diff
@@ -1000,6 +1003,26 @@ def build_pxe_default(self, synchronous=True, **kwargs):
...
+            ``True``. Immediately return the server's response otherwise.
+        :param kwargs: Arguments to pass to requests.
+        :returns: The server's response, with all JSON decoded.
+        :raises: ``requests.exceptions.HTTPError`` If the server responds with
+            an HTTP 4XX or 5XX message.
+        """
+        kwargs = kwargs.copy()  # shadow the passed-in kwargs
+        kwargs.update(self._server_config.get_client_kwargs())
+        response = client.post(self.path('clone'), **kwargs)
+        result = _handle_response(response, self._server_config, synchronous)
```
Comment (elyezer) [inline nailgun/entities.py]: In order to follow the other helper methods, this helper should return the response here and then the caller should get this information and parse as needed. By doing that we have more options from the outside in order to check proper response before going ahead and getting the cloned template.

Comment (abalakh) [inline nailgun/entities.py]: It was originally like that, i proposed this approach in the [comment above](https://github.com/SatelliteQE/nailgun/pull/300#discussion_r67849953) :smile: 
Sorry for that

**Comment by abalakh** [comment]
ACK

**Comment by elyezer** [comment]
ACK

**Pull request merged**